common/pathconverters/pathconverters.go

```diff
@@ -198,7 +199,17 @@ func ChangePaths(obj interface{}, mapping map[string]string) error {
 			logrus.Errorf("%s", err)
 			return path, err
 		}
-		for src, dest := range mapping {
+		keys := make([]string, 0, len(mapping))
+		for k := range mapping {
+			keys = append(keys, k)
+		}
+
+		sort.Slice(keys, func(i, j int) bool {
+			return len(keys[i]) > len(keys[j])
+		})
+
+		for _, src := range keys {
+			dest := mapping[src]
```

Comment:
This change was discussed with @kmehant. 

```
I have stumbled across a blocker for plan and then transform usecase. 
The problem was with the usage of temp directory to store the remote sources that are cloned. 
In the code here https://github.com/konveyor/move2kube/blob/6aee6e37fdb268e5cf0f25f35ffeba00291cc725/common/pathconverters/pathconverters.go#L191 
a function is used to convert absolute paths in the plan file to relative paths before being written into plan file. 
To this function both temp directory and sources directory are sent. 
The implementation tries to get the relative path with any of the provided parent paths to the paths in the plan file. 
In the case of using remote move2kube sources in the temp dir makes both of these directories as potential parents for the paths in the plan file. 
Due to this reason, I have worked on a workaround where we consider the longest parent paths first and then move to shorter ones. 
So that the most matching parents paths are chosen. 
This workaround worked for me.
```

Reply:
we might better iterate over all the paths and then choose the one have the highest prefix match in that case in the above example, the chosen path will be `/a/v/temp/anotherfolder` and relative path will be `.`

Reply:
lets get inputs from @ashokponkumar and @HarikrishnanBalagopal as well.